### PR TITLE
Automatically detect Resource Disk 

### PR DESCRIPTION
**Pull request by mearvind**

### CODE REVIEW

File: VMBackup/main/Utils/ResourceDiskUtil.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+									device = d.split(':')[1]
+									return device
+		except (OSError, IOError) as exc:
+			err_msg='Error getting device for %s or %s: %s , Stack Trace: %s' % (gen1_device_prefix, gen2_device_id, str(exc),traceback.format_exc())
+			self.logger.log(err_msg,True,'Error')
+		return None
+
+	def device_for_ide_port(self, port_id=1):
+		"""
+		Return device name attached to ide port 'n'.
```
Comment (soumyapattnaik) [inline VMBackup/main/Utils/ResourceDiskUtil.py:90]: put some more comments

File: VMBackup/main/Utils/ResourceDiskUtil.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+		"""
+		if port_id > 3:
+			return None
```
Comment (soumyapattnaik) [inline VMBackup/main/Utils/ResourceDiskUtil.py:93]: this code not required for temporary disk

File: VMBackup/VMBackup.pyproj [hunk 3]
```diff
@@ -23,7 +23,6 @@
   </PropertyGroup>
   <PropertyGroup>
     <VisualStudioVersion Condition="'$(VisualStudioVersion)' == ''">10.0</VisualStudioVersion>
-    <PtvsTargetsFile>$(MSBuildExtensionsPath32)\Microsoft\VisualStudio\v$(VisualStudioVersion)\Python Tools\Microsoft.PythonTools.targets</PtvsTargetsFile>
   </PropertyGroup>
   <ItemGroup>
     <Folder Include="main\" />
```
Comment (soumyapattnaik) [inline VMBackup/VMBackup.pyproj:28]: remove these files

File: VMBackup/main/Utils/SizeCalculation.py [hunk 4]
```diff
@@ -146,7 +168,7 @@ def get_total_used_size(self):
                     self.logger.log("Not Adding network-drive, Device name : {0} used space in KB : {1} fstype : {2}".format(device,used,fstype),True)
                     total_used_network_shares = total_used_network_shares + int(used)
 
-                elif device == '/dev/sdb1' and self.isOnlyOSDiskBackupEnabled == False : #<todo> in some cases root is mounted on /dev/sdb1
+                elif device == resource_disk_device and self.isOnlyOSDiskBackupEnabled == False : #<todo> in some cases root is mounted on /dev/sdb1
```
Comment (soumyapattnaik) [inline VMBackup/main/Utils/SizeCalculation.py:171]: revert this change and add a logger to find how much under billing we are doing

File: VMBackup/main/Utils/SizeCalculation.py [hunk 5]
```diff
@@ -51,6 +52,17 @@ def get_loop_devices(self):
...
                 disk_loop_devices_file_systems.append(file_system_info[0])
         return disk_loop_devices_file_systems
+  
+    def device_list_for_billing(self):
+        self.logger.log("In device_list_for_billing",True)
+        devices_to_bill = [] #list to store device names to be billed
+        device_items = disk_util.get_device_items(None)
+        for device_item in device_items :
+            # self.logger.log("Device name : {0} ".format(str(device_item.name)),True)
+            if str(device_item.name).startswith("sd"):
```
Comment (soumyapattnaik) [inline VMBackup/main/Utils/SizeCalculation.py:62]: check for 32 disks

Comment (soumyapattnaik) [reply VMBackup/main/Utils/SizeCalculation.py:62]: if tested for 32 disks. @mearvind  please mark it resolved

Comment (mearvind) [reply VMBackup/main/Utils/SizeCalculation.py:62]: I asked for it from Arpit but did not get it till now

**Review by soumyapattnaik — APPROVED** [review]
